CRM/Yoteup/Form/Report/IndividualCounseller.php

```diff
@@ -300,12 +325,56 @@ function postProcess() {
   }
   
   function createTemp() {
-    $sql = "CREATE TABLE civicrm_watchdog_temp AS SELECT {$this->_aliases['civicrm_contact']}.id, p.purl_145, MIN(DATE(FROM_UNIXTIME(timestamp)))
+    $sqlA = "CREATE TEMPORARY TABLE civicrm_watchdog_temp_a AS
+      SELECT wid, SUBSTRING_INDEX(SUBSTRING_INDEX(location, '://', -1), '.', 1) as purl, MIN(DATE_FORMAT(DATE(FROM_UNIXTIME(timestamp)),'%m/%d/%Y')) as first_visit
+      FROM {$this->_drupalDatabase}.watchdog
+      GROUP BY SUBSTRING_INDEX(SUBSTRING_INDEX(location, '://', -1), '.', 1)";
+    $dao = CRM_Core_DAO::executeQuery($sqlA);
+    
+    $sql = "CREATE TEMPORARY TABLE civicrm_watchdog_temp_b AS
+      SELECT {$this->_aliases['civicrm_contact']}.id, p.purl_145, first_visit
       FROM civicrm_contact {$this->_aliases['civicrm_contact']}
       INNER JOIN civicrm_value_nrmpurls_5 p ON {$this->_aliases['civicrm_contact']}.id = p.entity_id
-      INNER JOIN {$this->_drupalDatabase}.watchdog w ON (SUBSTRING_INDEX(SUBSTRING_INDEX(location, '://', -1), '.', 1)) = p.purl_145";
-    CRM_Core_Error::debug( '$sql', $sql );
-    exit;
+      INNER JOIN civicrm_watchdog_temp_a w ON w.purl = p.purl_145 COLLATE utf8_general_ci";
+    $dao = CRM_Core_DAO::executeQuery($sql);
+
+    $sql = "CREATE TEMPORARY TABLE civicrm_watchdog_temp_c AS
+      SELECT cc.id, MAX(ws.sid), GROUP_CONCAT(wsd22.data, ' ', wsd23.data, ' ', wsd24.data) as brochures
+      FROM civicrm_contact cc
+      LEFT JOIN {$this->_drupalDatabase}.webform_submitted_data wsd ON wsd.data = cc.id AND wsd.cid = 2
+      LEFT JOIN {$this->_drupalDatabase}.webform_submitted_data wsd22 ON wsd22.cid = 22 AND wsd22.sid = wsd.sid
+      LEFT JOIN {$this->_drupalDatabase}.webform_submitted_data wsd23 ON wsd23.cid = 23 AND wsd23.sid = wsd.sid
+      LEFT JOIN {$this->_drupalDatabase}.webform_submitted_data wsd24 ON wsd24.cid = 24 AND wsd24.sid = wsd.sid
+      LEFT JOIN {$this->_drupalDatabase}.webform_submissions ws ON ws.sid = wsd.sid
+      WHERE ws.nid = 72
+      GROUP BY cc.id";
```

Comment:
@JoeMurray I need help with your expertise on sql here. I'm not sure if this is the right way to be doing the group concat since there are 2x or even 3x values for each component. Let's discuss this on skype.